src/LineBot/Events/LineEventCollection.cs

```diff
@@ -12,13 +12,35 @@
 // License for the specific language governing permissions and limitations
 // under the License.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
 using Newtonsoft.Json;
 
 namespace Line
 {
-    internal sealed class LineEventCollection
+    [JsonObject]
+    internal sealed class LineEventCollection : ILineEvents
     {
+        [JsonProperty("destination")]
+        string ILineEvents.Destination { get; set; }
+
         [JsonProperty("events")]
-        public LineEvent[] Events { get; set; }
+        List<ILineEvent> ILineEvents.Events { get; set; }
+
+        public IEnumerator<ILineEvent> GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return (this as ILineEvents).GetEnumerator();
+        }
+
+        internal static ILineEvents Empty()
+        {
+            return new LineEventCollection();
```

Comment:
You are now returning a `LineEventCollection` where the `Events`  property is null. Should be changed to:
```C#
return new LineEventCollection()
{
    Events = new List<ILineEvent>()
};
```